hyperspy/signal_tools.py

```diff
@@ -238,7 +243,162 @@ def apply(self):
+        else:
+            self.edges_list = ()
+            energy, relevance, description = (), (), ()
+
+        self._keep_valid_edges()
+
+        return self.edges_list, energy, relevance, description
+
+    def _keep_valid_edges(self):
+        edge_all = list(self.signal._edge_markers.keys())
+        for edge in edge_all:
+            if (edge not in self.edges_list):
+                if edge in self.active_edges:
+                    self.active_edges.remove(edge)
+                elif edge in self.active_complementary_edges:
+                    self.active_complementary_edges.remove(edge)
+                self.signal.remove_EELS_edges_markers([edge])
+            elif (edge not in self.active_edges):
+                self.active_edges.append(edge)
+        
+        self.on_complementary()
+        self._plot_labels()
+
+    def update_active_edge(self, change):
+        state = change['new']
+        edge = change['owner'].description
+
+        if state:
+            self.active_edges.append(edge)
+        else:
+            if edge in self.active_edges:
+                self.active_edges.remove(edge)
+            if edge in self.active_complementary_edges:
+                self.active_complementary_edges.remove(edge)
+            self.signal.remove_EELS_edges_markers([edge])
+            
+        self.on_complementary()
+        self._plot_labels()
+
+    def on_complementary(self):
+        
+        if self.complementary:
+            self.active_complementary_edges = \
+                self.signal.get_complementary_edges(self.active_edges, 
+                                                    self.only_major)
+        else:
+            self.active_complementary_edges = []
+
+    def check_btn_state(self):
+        
+        edges = [btn.description for btn in self.btns]
+        for btn in self.btns:
+            edge = btn.description
+            if btn.value is False:
+                if edge in self.active_edges:
+                    self.active_edges.remove(edge)
+                    self.signal.remove_EELS_edges_markers([edge])                
+                if edge in self.active_complementary_edges:
+                    btn.value = True
+                    
+            if btn.value is True and self.complementary:
+                comp = self.signal.get_complementary_edges(self.active_edges, 
+                                                           self.only_major)
+                for cedge in comp:
+                    if cedge in edges:
+                        pos = edges.index(cedge)
+                        self.btns[pos].value = True
+
+    def _plot_labels(self, active=None, complementary=None):
+        # plot selected and/or complementary edges
+        if active is None:
+            active = self.active_edges
+        if complementary is None:
+            complementary = self.active_complementary_edges
+            
+        self._clear_all_markers()
```

Comment:
Oh I haven't checked the qt backend! The marker's positions are now updated instead of removing them first and then adding them, this gives a smoother experience.